### PR TITLE
Add networking cables

### PR DESCRIPTION
**Pull request by Illiux**
## About the PR 

This adds networking cables to the game. I thought they'd be good as a PR on their own because it turns out there's quite a bit of scattered modifications required to implement a new kind of cable, and they're functional (despite not being useful).

**Screenshots**

netcable
netcable-lit

**Changelog**

:cl: illiux
- add: Networking cables. These can be placed similar to power cables, but are not currently used for anything more.

### CODE REVIEW

**Comment by Just-a-Unity-Dev** [comment]
What does this add to game? This seems like unused content

File: Resources/Prototypes/Entities/Objects/Tools/cable_coils.yml [hunk 1]
```diff
@@ -131,3 +131,40 @@
...
+
+- type: entity
+  parent: CableNetStack
+  id: CableNetStack1
+  suffix: 1
+  components:
+  - type: Sprite
+    state: coilnet-10
+  - type: Item
+    state: 3
```
Comment (Just-a-Unity-Dev) [inline Resources/Prototypes/Entities/Objects/Tools/cable_coils.yml]: Do you mean `sprite`?

**Review by Just-a-Unity-Dev — CHANGES_REQUESTED** [review]

**Comment by Illiux** [comment]
It doesn't add anything at the moment, but the alternative is bundling it with some later change which makes for a large, hard to review PR. Particularly with how scattered the changes here are, that'd really hurt the readability of whatever other potential PR it got bundled with.

File: Resources/Prototypes/Entities/Objects/Tools/cable_coils.yml (hunk 1 above)
Comment (Just-a-Unity-Dev) [reply Resources/Prototypes/Entities/Objects/Tools/cable_coils.yml]: Nevermind, your latest commit (1a3a4649083ad063dc8f0489493ec2955ceaf896) fixed it

**Comment by Just-a-Unity-Dev** [comment]
I don't see the need for this as we already have device net

**Comment by Partmedia** [comment]
> It doesn't add anything at the moment, but the alternative is bundling it with some later change which makes for a large, hard to review PR. Particularly with how scattered the changes here are, that'd really hurt the readability of whatever other potential PR it got bundled with.

It would probably be helpful if you gave an example of planned content that uses these cables, and explain why what already exists doesn't cut it.

**Comment by mirrorcult** [comment]
we aren't going to add stuff based on a hypothetical future feature that we know nothing about. i also personally dislike the concept of having a separate cable entirely for networking--there are already solutions ingame for this (apc devicenet). heavy justification is needed for this

**Review by mirrorcult — CHANGES_REQUESTED** [review]
see above

**Comment by juliangiebel** [comment]
I assume those cables are supposed to be used instead of having the wired network use power cables or for a new connection type?
I don't think having even more cables is a good idea. Also wired network currently doesn't use power cables. It only requires the devices to be on the same grid because of performance reasons.

**Comment by vulppine** [comment]
there is some precedence for these, and it involves a discussion about an implementation about wired device networks as something *technically* more cable bound - this occurred in discord in the development channel (starts around https://discord.com/channels/310555209753690112/310555209753690112/1017982776512888932, where i discuss about the current implementations that use devicenet as the backend)

i think there was a discussion in the past about having a separate cable type for networking, but this solution that we discussed seems somewhat interesting: maybe posting a more collected version of this would be more helpful, and combining this PR with a working implementation would be more advised

**Comment by Illiux** [comment]
I've written up a high-level design of the wider system I had in mind: https://hackmd.io/@K6FF-6JhTGSCAS672tDDbw/SJhEAh6li

With more consideration I decided that the tradeoffs actually favor using the powernet, so I've managed to eliminate a new cable type from the design. Accordingly, I'm closing this PR.

**Pull request closed without merging**